aiida/tools/dbimporters/plugins/matp.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module that contains the class definitions necessary to offer support for
+queries to Materials Project."""
+
+from __future__ import absolute_import
+import os
+import datetime
+import requests
+
+from pymatgen import MPRester
+from aiida.tools.dbimporters.baseclasses import CifEntry, DbEntry, DbImporter, DbSearchResults
+
+
+class MatProjImporter(DbImporter):
+    """
+    Database importer for the Materials Project.
+    """
+
+    _collection = 'structures'
+    _supported_keywords = None
+
+    def __init__(self, api_key=None):
+        """
+        Instantiate the MatProjImporter by setting up the Materials API (MAPI) connection details
+
+        :param api_key: the API key to be used to access MAPI
+        """
+        self.setup_db(api_key=api_key)
+        self._mpr = MPRester(self._api_key)
+
+    def setup_db(self, **kwargs):
+        """
+        Setup the required parameters to the REST API
+
+        :param api_key: the API key to be used to access MAPI
+        """
+        api_key = kwargs['api_key']
+        if api_key is None:
+            try:
+                self._api_key = os.environ['PMG_MAPI_KEY']
```

Comment:
Replace
```Python
self._api_key = os.environ['PMG_MAPI_KEY']
```
with
```Python
api_key = os.environ['PMG_MAPI_KEY']
```
Otherwise the ``self._api_key`` will be overwritten with ``None``.

Reply:
Ouch. Thanks. Fixed.